server/athenian/api/internal/features/metric_calculator.py

```diff
@@ -1007,6 +1008,69 @@ def group_by_lines(lines: Sequence[int], column: np.ndarray) -> list[np.ndarray]
     return full_line_groups
 
 
+@dataclasses.dataclass(frozen=True, slots=True)
+class JIRAGrouping:
+    """Grouping definition based on jira entities.
+
+    `None` field values indicate that no grouping should be done on the corresponding dataframe
+    property.
+    On the opposite an single empty field value makes this group select nothing.
+    """
+
+    projects: Optional[Collection[str]] = None
+    priorities: Optional[Collection[str]] = None
+    types: Optional[Collection[str]] = None
+
+    def __bool__(self) -> bool:
+        """Get whether the group definition is empty.  An empty group does not do any grouping."""
+        return any(f is not None for f in (self.projects, self.priorities, self.types))
+
+    @classmethod
+    def empty(cls) -> JIRAGrouping:
+        """Return a `JIRAGrouping` selecting all data."""
+        return cls(None, None, None)
```

Comment:
No longer needed, we can call `JIRAGrouping()` directly.

Reply:
https://github.com/athenianco/athenian-api/pull/2803#discussion_r972079702